### ISSUE TITLE
no strain displayed in phenotype annotation summary table

### ISSUE THREAD

**Issue by CuzickA** [pombase/canto#1691]
Looks like strain display is missing from several places

Metagenotype pathogen and host pickers including wildtype host organism with no genes
Metagenotype table

All phenotype annotation summary tables

Not sure if this has been covered in other tickets but it is really important

**Comment by barnabynorman** [comment]
@CuzickA to be clear, for the phenotype annotation, are they always using alleles? Can you describe the process to create one? I have only been able to create what looks like annotations with genes??

**Comment by barnabynorman** [comment]
@kimrutherford I have looked at this but it looks like it is being generated from perl.

I created the annotation as per @CuzickA's instructions and can see it at:

```
/curs/08aa578a31d2f234/feature/metagenotype/view/6
```

I traced the code back to:

```
root/static/ng_templates/annotation_table_ontology_row.html
```

On line 8 it is outputting a parameter of annotation:

```
metagenotype_display_name
```

which is being generated in:

```
lib/Canto/Curs/Utils.pm
```

on line 126.

No idea where to get the strain info from to add it to the display name - I was thinking that for each genotype listed I would put the strain in brackets:

> Fusarium graminearum TRI5Δ (PH1) / Triticum aestivum wild type (Bobwite)

**Comment by kimrutherford** [comment]
> Metagenotype pathogen and host pickers including wildtype host organism with no genes

Do the pickers need to show the strains since the strain appears in the tables?

> Metagenotype table

I think that's fixed?

**Comment by jseager7** [comment]
> Should Pathogen genotype be the first column?

I'd say **yes**, since I think this is the convention we've used on the metagenotype page.

> Metagenotype pathogen and host pickers including wildtype host organism with no genes
>> Do the pickers need to show the strains since the strain appears in the tables?

I think this is possibly talking about the fact that strains weren't shown in the genotype tables in some cases, possibly because of the bug from #1703. Currently the strains do show in all tables (including the 'wildtype' for hosts with no genes), but there's a bug with the fact that the strain rows aren't merged for multi-allele genotypes:

![mg_multi_strain_bug](https://user-images.githubusercontent.com/37659591/49514853-32228380-f88d-11e8-94ca-5dd8e446c306.PNG)

I think this is because the metagenotype page uses a different template, `genotype_simple_list_view.html` (which is unique to the metagenotype page), instead of re-using the `genotype_list_view.html` template from the genotype management page. So my fix in #1700 possibly isn't having an effect here.

**Comment by barnabynorman** [comment]
> I think this is because the metagenotype page uses a different template, genotype_simple_list_view.html (which is unique to the metagenotype page), instead of re-using the genotype_list_view.html template from the genotype management page. So my fix in #1700 possibly isn't having an effect here.

@jseager7 you would need to duplicate any work done in genotype rows and views as the metagenotype ones diverged when coding the metagenotypes, (as I remember) some dependencies in the old code could not be fulfilled - so I started by simplifying them to make it work - its not ideal but at the time seemed like the simplest solution. Given more time I would re-factor the genotype code to do both.

They may look alike and share common features but its now different code!

**Comment by jseager7** [comment]
OK, thanks for the information. For now I've just made it so the strain rows in `genotype_simple_list_row.html` have a `rowspan` applied in the same way as `genotype_list_row.html`. I'll move the refactoring task to a new issue.
